src/server/game/AI/SmartScripts/SmartAI.cpp

```diff
@@ -727,13 +712,18 @@ void SmartAI::OnCharmed(bool apply)
     {
         if (HasEscortState(SMART_ESCORT_ESCORTING | SMART_ESCORT_PAUSED | SMART_ESCORT_RETURNING))
             EndPath(true);
```

Comment:
is this correct? whenver someone mind controls ur escorting creature u fail the quest?